.github/workflows/update-flake-dependencies.yml

```diff
@@ -0,0 +1,30 @@
+name: Update flake dependencies
+
+on:
+  schedule:
+    - cron: '0 16 * * 5'
+  workflow_dispatch: # for allowing manual triggers of the workflow
+
+jobs:
+  update-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: DeterminateSystems/nix-installer-action@main
+    - uses: DeterminateSystems/magic-nix-cache-action@main
+    - name: update flake.lock
+      run: nix flake update
+    - name: Create signed commit with flake.lock changes
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        FILE_TO_COMMIT: flake.lock
+        COMMIT_BRANCH: automation/update-flake-dependencies
+        COMMIT_MESSAGE: "chore(nix): Update Flake dependencies"
+      run: |
+          # commit via the GitHub API so we get automatic commit signing
+          gh api --method PUT /repos/1Password/shell-plugins/contents/$FILE_TO_COMMIT \
+            --field message="$COMMIT_MESSAGE" \
+            --field content=@<(base64 -i $FILE_TO_COMMIT) \
+            --field branch="$COMMIT_BRANCH" \
+            --field sha="$(git rev-parse $COMMIT_BRANCH:$FILE_TO_COMMIT)"
+          gh pr create --title "[automation]: Update Flake dependencies" --reviewer mrjones2014 --base main --head $COMMIT_BRANCH
```

Comment:
I got this from here: https://gist.github.com/swinton/03e84635b45c78353b1f71e41007fc7c

I'm not 100% sure if this will work, and I'm not sure how to test it other than merging it to `main` and running the job. Let me know if we'd rather punt on this and tackle this in a follow-up PR.

Reply:
I may be able to make something like that work. I'm not sure how to downgrade the flake.lock (other than manually getting all the versions and hashes and such) but I think it would also work if we outright delete flake.lock as a test. I'll try this out today.

Reply:
Tested by deleting `flake.lock`. I've pushed a couple changes and it works now, you can see in #436 that the latest commit on that PR is the automated commit from the bot and it is indeed a singed commit! Yay for automated `flake.lock` updates!